libtiledbsoma/src/utils/arrow_adapter.h

```diff
@@ -257,6 +257,37 @@ class ArrowAdapter {
         ArrowTable index_column_info,
         std::string soma_type,
         bool is_sparse = true,
+        PlatformConfig platform_config = PlatformConfig(),
+        ArrowTable spatial_column_info = {
+            std::unique_ptr<ArrowArray>(nullptr),
+            std::unique_ptr<ArrowSchema>(nullptr)});
+
+    /**
+     * @brief Get a TileDB attribute with its enumeration from an Arrow schema.
+     *
+     * @return std::pair<Attribute, std::optional<Enumeration>>
+     */
+    static std::pair<Attribute, std::optional<Enumeration>>
+    tiledb_attribute_from_arrow_schema(
+        std::shared_ptr<Context> ctx,
+        ArrowSchema* arrow_schema,
+        std::string_view type_metadata,
+        PlatformConfig platform_config = PlatformConfig());
+
+    /**
+     * @brief Get a TileDB dimension from an Arrow schema.
+     *
+     * @return std::pair<Dimension, bool> The TileDB dimension with a boolean
+     * flag indicating whether or not the dimension uses `current domain`.
+     */
+    static std::pair<Dimension, bool> tiledb_dimension_from_arrow_schema(
```

Comment:
i'd prefer you move the declaration of this dimension function above the corresponding attribute function